### PR TITLE
Merge 4.4.0 release to main

### PR DESCRIPTION
**Pull request by vipulnsward**
## Description

- Add changelog for upcoming release
- Rebase on top of existing commits from 4.4.0 branch and main
- Will draft a new release once this looks okay

## Checklist

- [x] Tests (if applicable)
- [x] Documentation (if applicable)
- [ ] Changelog stub (or use [conventional commit messages](https://www.conventionalcommits.org/))

### CODE REVIEW

File: spec/uploadcare/entity/uploader_spec.rb [hunk 1]
```diff
@@ -2,106 +2,131 @@
 
 require 'spec_helper'
 
-module Uploadcare
```
Comment (vipulnsward) [inline spec/uploadcare/entity/uploader_spec.rb:5]: These are mostly whitespace changes. Better diff view available here: https://github.com/uploadcare/uploadcare-ruby/pull/156/files?diff=split&w=1#diff-927717518ecbedcd1649c3a325798166ecf1e10b4736a26820fb09170f211505

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,23 @@
 # Changelog
 
+## 4.4.0 — 2024-03-09
+
+### Breaking 
+
+* Drop support of unmaintainable Ruby versions < 3.x
```
Comment (rsedykh) [inline CHANGELOG.md:7]:
```suggestion
* Drop support of unmaintainable Ruby versions < 3.x.
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,23 @@
...
+
+### Fixed
+
+* Update locations where Dry::Monads structure has changed.
+* Sign URL uploads if configured (#139)
```
Comment (rsedykh) [inline CHANGELOG.md:12]:
```suggestion
* Sign URL uploads if configured (#139).
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,23 @@
...
+* Start returning proper error message when raising RequestError in poll_upload_response, to hint to users what is going on. Fixes #141
```
Comment (rsedykh) [inline CHANGELOG.md:13]:
```suggestion
* Start returning proper error message when raising RequestError in poll_upload_response, to hint to users what is going on. Fixes #141.
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,23 @@
...
+* When polling, raise if an error is returned (#142)
```
Comment (rsedykh) [inline CHANGELOG.md:14]:
```suggestion
* When polling, raise if an error is returned (#142).
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,23 @@
...
+* Fix documentation about original file url on simple file upload.
+
+### Changed
+* Support params in Rest client and in file info method, to allow passing custom params like "include=appdata" in `Uploadcare::File.file` calls. Closes #132
```
Comment (rsedykh) [inline CHANGELOG.md:18]:
```suggestion
* Support params in Rest client and in file info method, to allow passing custom params like "include=appdata" in `Uploadcare::File.file` calls. Closes #132.
```

**Review by rsedykh — APPROVED** [review]
Added dots. Just feels right for me. :-)

**Pull request merged**